### PR TITLE
DR-596 Add unit test workflow

### PR DESCRIPTION
**Pull request by timrickwood**
## Description
Add reusable unit test workflow and python unit test action

## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [ ] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by stuartreid — APPROVED** [review]
LGTM

File: .github/workflows/unit_test.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+      - name: "Checkout code"
+        uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@main
+      - name: Configure AWS Credentials
+        uses: NHSDigital/uec-dos-management/.github/actions/configure-credentials@main
+        with:
+          aws_account_id: ${{ secrets.ACCOUNT_ID }}
+          aws_region: ${{ vars.AWS_REGION }}
+      - name: "Run python unit test"
+        uses: NHSDigital/uec-dos-management/.github/actions/unit-test-python@task/DR-5964_Add_unit_test_workflow
```
Comment (jope1) [inline .github/workflows/unit_test.yaml:31]: Don't forget to change back to main

File: .github/actions/unit-test-python/action.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  aws_region:
+    description: "AWS region"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install python
+      id: install-python
+      uses: actions/setup-python@v4
```
Comment (jope1) [inline .github/actions/unit-test-python/action.yaml:23]: Do we want a wrapper action here?

Comment (timrickwood) [reply .github/actions/unit-test-python/action.yaml:23]: ooh good shout

**Review by jope1 — APPROVED** [review]

**Review by tamaragoldschmidt — APPROVED** [review]

**Pull request merged**